### ISSUE TITLE
need help: Promise.withResolvers is not a function

### ISSUE THREAD

**Issue by DidoFitch** [TaTo30/vue-pdf#127]
Hello, this issue is not a bug but I need some help. 

I‘m using **version 1.10.0 of this repository**, and in **older versions of Chrome (earlier than 105)**, Promise.withResolvers is not supported, causing an error when I preview PDFs: Promise.withResolvers is not a function. 

I tried using Babel to support it or ask ChatGPT, but it didn't work. Can anyone help me?

![Snipaste_2024-07-10_09-47-39](https://github.com/TaTo30/vue-pdf/assets/27599714/7af50486-4db7-4923-b44e-986e894ab428)

**I have tried this:**

yarn add @babel/core @babel/preset-env @vitejs/plugin-vue vite-plugin-babel -D

yarn add promise-polyfill

vite.config.js:

```js
import { defineConfig } from 'vite';
import vue from '@vitejs/plugin-vue';
import viteBabel from 'vite-plugin-babel';

export default defineConfig({
  plugins: [
    vue(),
    viteBabel()
  ]
});
```
main.js:
```js
import 'promise-polyfill/src/polyfill';
if (!Promise.withResolvers) {
  Promise.withResolvers = function() {
    let resolve, reject;
    const promise = new Promise((res, rej) => {
      resolve = res;
      reject = rej;
    });
    return { promise, resolve, reject };
  };
}

```

**Comment by starter-dev** [comment]
Hi. Thank you for your post! I noticed that you mentioned it's not actually an issue. It seems like your topic would be a great fit for the "Discussions" section instead. Discussions have a Q&A section that is perfect for this type of content, and you'll likely get more engagement there.

**Comment by TaTo30** [comment]
After checking this I found this problem on two points:
![imagen](https://github.com/user-attachments/assets/f51de620-005d-48a9-9537-94f328d03929)

The first one is from `pdf.js` file and can be patched using a polyfill or the code @DidoFitch shared.

The second one is from the `pdf.worker.js` and cannot be polyfilled surely because the worker is running on different thread and context. In this library the worker is embebed directly:
https://github.com/TaTo30/vue-pdf/blob/de24ae408d67cf892cb1e51bf19119056ba4629e/src/components/composable.ts#L2

Using the legacy worker fix the second point but It's still required to add the polyfill for the first point.
```js
import PDFWorker from 'pdfjs-dist/legacy/build/pdf.worker.min?url'
```

**Comment by TaTo30** [comment]
This should works, It's important configure worker before using `usePDF`:

```diff
<script setup lang="ts">
+ import * as PDFJS from 'pdfjs-dist';
+ import LegacyWorker from 'pdfjs-dist/legacy/build/pdf.worker.min?url';
import { VuePDF, usePDF } from '@tato30/vue-pdf';

+ PDFJS.GlobalWorkerOptions.workerSrc = LegacyWorker

const { pdf } = usePDF('https://raw.githubusercontent.com/mozilla/pdf.js/ba2edeae/web/compressed.tracemonkey-pldi-09.pdf')
</script>

<template>
  <div>
    <VuePDF :pdf="pdf" />
  </div>
</template>
```

**Comment by CryptoNinja1210** [comment]
Hi TaTo!
I need you help
Could you tell me how can I fix this deployment issue?
`Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "application/octet-stream". Strict MIME type checking is enforced for module scripts per HTML spec.`

**Comment by CryptoNinja1210** [comment]
I am sorry to border you.
I fixed this problem.
`# Nginx configuration
events{}

http {

    default_type application/octet-stream;
    include  /etc/nginx/mime.types;
    # include the server config for your application
    include /etc/nginx/app.conf;

    types {
        text/javascript                 js;
        application/javascript          mjs;
    }
}`
I updated nginx.conf file.
I leave code of nginx.conf file for other developers
Thank you for your time.

**Comment by ZumelzuR** [comment]
There area PR for this?

**Comment by TaTo30** [comment]
@ZumelzuR Nope, this issue can be fixed using https://github.com/TaTo30/vue-pdf/issues/127#issuecomment-2227189730 

It is not necessary to do something on source code.

**Comment by ZumelzuR** [comment]
> @ZumelzuR Nope, this issue can be fixed using [#127 (comment)](https://github.com/TaTo30/vue-pdf/issues/127#issuecomment-2227189730)
> 
> It is not necessary to do something on source code.

I saw that there are some code in master (that is doing this PDFJS.GlobalWorkerOptions.workerSrc = LegacyWorker) but is there is not a compiled version of this. Would be great have this master changes on a compiled version :)
